torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2529,6 +2529,12 @@ such as `dist.all_reduce(tensor, async_op=True)`.
       .def(
           "result",
           [](::c10d::Work& work) -> std::vector<at::Tensor> {
+            // Deprecation reason:
+            // Work.result() returns a vector of tensors. This signature is
+            // problematic as some collectives may just return one tensor (e.g
+            // all-reduce), while some others may return multiple tensors (e.g.
+            // all-gather).
```

Comment:
should we also mention the memory lifetime issue?

Reply:
Done!